libc-bottom-half/crt/crt1-command.c

```diff
@@ -3,8 +3,20 @@ extern void __wasm_call_ctors(void);
 extern int __main_void(void);
 extern void __wasm_call_dtors(void);
 
+// Commands should only be called once per instance. This simple check ensures
+// that the `_start` function isn't started more than once. `address_space(1)`
+// tells clang to put this variable in a wasm global.
+__attribute__((address_space(1)))
+static volatile int started = 0;
+
 __attribute__((export_name("_start")))
 void _start(void) {
+    // Don't allow the program to be called multiple times.
+    if (started != 0) {
+	__builtin_trap();
+    }
+    started = 0;
```

Comment:
Seems like it would be good place for `#ifndef NDEBUG` but we don't have a debug version of libc yet IIUC

Reply:
I care about code size, but this is just a few bytes. And this hazard comes up in both component use cases and JS embedding use cases, where creating an instance is separated from running the command and there's no other way to prevent the command from being run multiple times, causing UB.

Reply:
`started = 1;` here